compiler-rt/test/asan/TestCases/Windows/dll_intercept_memset.cpp

```diff
@@ -1,9 +1,9 @@
-// RUN: %clang_cl_asan %Od %p/dll_host.cpp %Fe%t
-// RUN: %clang_cl_asan -Wno-fortify-source %LD %Od %s %Fe%t.dll
+// RUN: %clang_cl_asan %Od /Oy- %p/dll_host.cpp %Fe%t
```

Comment:
Don't use an unconditional `/Oy-` here - we run these tests with mingw compilers as well, with arguments replaced with corresponding ones for the GNU style compiler driver - hence substitutions like `%Od` and similar here; see `test/asan/lit.cfg.py`.

Reply:
I think I actually have the fixes for this in a follow-up.

I'll append that commit here and we can merge both at once if you'd like.

I could alternatively create another (stacked) PR with those changes

Reply:
Hmm, depending on how big that extra change is, it may be ok to add that change here. But this change is in itself already quite big, so it could also be clearer to just back out this change from this particular PR (so it doesn't fix everything for MSVC, just the bulk of things) and then do that change in a separate step afterwards, for the sake of easier readability/reviewability.

Reply:
Turns out we hadn't fixed this internally, but I've added it here.

I decided to just add the substitution, although using `%if` could also make sense considering this is the only place where /Oy- shows up.